src/app.js

```diff
@@ -0,0 +1,96 @@
+const app = angular.module('myApp', []);
+app.controller('myController', ($scope) => {
+  /**
+  * Returns a message
+  * @param {string} message - a message to be displayed to user
+  * @return {string} returns a string message
+  */
+  function showMessage(message) {
+    $scope.message = message;
+    $('#response-modal').modal();
+    $scope.$apply();
+  }
+  $scope.error = '';
+  $scope.invertedIndex = new InvertedIndex();
+  //  Make search input file and button hidden
+  $scope.display = false;
+  //  Get a list names of the uploaded files
+  $scope.uploaded = [];
+  // get a list of documents in a file
+  $scope.tableHeader = {};
+  /**
+  * Checks if a file is a valid json file
+  * @param {object} file - file contents
+  * @return {boolean} returns a boolean or string
+  */
+  function checkJson(file) {
+    if (file.name.split('.')[1] !== 'json') {
+      showMessage('invalid json file');
+      return;
+    }
+    return true;
+  }
+  //  Get the uploaded file
+  $scope.getFile = (file) => {
+    checkJson(file);
+    $scope.error = '';
+    const reader = new FileReader();
+    reader.onloadend = (e) => {
+      try {
+        $scope.invertedIndex.uploadedFiles[file.name] = JSON.parse(e.target.result);
+        if (!$scope.invertedIndex.uploadedFiles[file.name][0]) {
```

Comment:
Is this not a check for emptiness? If it is then it should be refactored, you can use `.length` and the message shown should depict emptiness